test/snippet/core/algorithm/for_each_type.cpp

```diff
@@ -0,0 +1,46 @@
+#include <iostream>
+#include <string>
+
+#include <seqan3/core/algorithm/parameter_pack.hpp>
+#include <seqan3/io/stream/debug_stream.hpp>
+
+using namespace seqan3;
+
+int main()
+{
+    // With c++20 you could also write it like this
+    // auto fn = []<typename type>(std::type_identity<type>)
+    // {
+    // ...
+    // };
+    auto fn = [](auto id)
+    {
+        using type = typename decltype(id)::type;
+
+        // id is of type std::type_identity<type>
+        using id_t = decltype(id);
```

Comment:
move this above of the type definition and then use
```cpp
using type = typename id_t::type;
```
I think it is easier to see the connection then